src/txn/txn.c

```diff
@@ -440,11 +440,37 @@ __wt_txn_config(WT_SESSION_IMPL *session, const char *cfg[])
 	WT_RET(__wt_config_gets_def(session, cfg, "read_timestamp", 0, &cval));
 	if (cval.len > 0) {
 #ifdef HAVE_TIMESTAMPS
+		bool round_to_oldest;
+		int cmp;
 		wt_timestamp_t ts;
+		WT_TXN_GLOBAL *txn_global;
 
+		txn_global = &S2C(session)->txn_global;
 		WT_RET(__wt_txn_parse_timestamp(session, "read", &ts, &cval));
-		WT_RET(__wt_timestamp_validate(session,
-		    &ts, &cval, true, false, false));
+		WT_WITH_TIMESTAMP_READLOCK(session, &txn_global->rwlock,
+		    cmp = __wt_timestamp_cmp(&ts,
+		    &txn_global->oldest_timestamp));
+		if (cmp < 0) {
+			/*
+			 * If given read timestamp is earlier than oldest
+			 * timestamp then round the read timestamp to
+			 * oldest timestamp based on round_to_oldest option.
+			 */
+			WT_RET(__wt_config_gets_def(session, cfg,
+			    "round_to_oldest", 0, &cval));
+			round_to_oldest = cval.val;
+			if (round_to_oldest) {
+				__wt_verbose_timestamp(session, &ts,
+				"Rounded to oldest timestamp");
```

Comment:
Rounded what?
Would "Rounded read timestamp to oldest timestamp" be more descriptive ?

Reply:
The output will be time Timestamp : <timestamp val> : Rounded to oldest timestamp

Reply:
Let's say timestamp value is 56, the message will be for instance "Timestamp 56 : Rounded to oldest timestamp".
Would it make more sense to say something like:
"Timestamp 56 : Rounded read timestamp to the oldest timestamp"

It conveys that a "read timestamp" got rounded and points to coming from the configure transaction semantics.